resources/manage-user

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+'''Manage users for repository access through basic authentication.'''
+
+from pathlib import Path
+import subprocess
+import argparse
+
+
+AUTH_FILE = Path(__file__).parent.parent.absolute() / 'basic-auth'
+
+
+class UnknownUser(Exception):
+
+    def __init__(self, user):
+        super().__init__('Unknown user "{}"'.format(user))
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(description=__doc__)
+    action = subparsers = parser.add_subparsers(
+        dest='action', help='action to perform')
+    action.required = True
+    add_action = subparsers.add_parser(
+        'add', help='add  a user. If the user exists, the password is updated')
+    add_action.add_argument('user', help='the username to add')
+    add_action.add_argument('password', help='the password for the user')
+    subparsers.add_parser('list', help='list users')
+    remove_action = subparsers.add_parser('remove', help='remove a user')
+    remove_action.add_argument('user', help='the user to remove')
+    return parser
+
+
+def add_user(auth_file, user, passwd):
+    '''Add a user.'''
+    auth_file.touch()  # ensure the file exists
+    subprocess.check_call(
+        ['htpasswd', '-b', str(auth_file), user, passwd],
```

Comment:
Hmm, let's not use the default password encryption of MD5 (!), how about `-B` for bcrypt?